File: lib/commonAPI/coreapi/ext/platform/android/src/com/rho/nativetabbar/NativeTabbarSingleton.java
package com.rho.nativetabbar;

import java.util.LinkedList;
import java.util.List;
import java.util.Map;

import com.rhomobile.rhodes.NativeBar;
import com.rhomobile.rhodes.api.IMethodResult;
import com.rhomobile.rhodes.util.Utils;

class NativeTabbarSingleton extends NativeTabbarSingletonBase implements INativeTabbarSingleton {
    public NativeTabbarSingleton(NativeTabbarFactory factory) {
        //super(factory);
    }

    public void create(List<Map<String, Object>> tabElements, Map<String, Object> tabBarProperties, IMethodResult result) {
    	if (result != null) {
    		Utils.platformLog("TABBAR", "create() result callback is not null !");
    	}
    	else {
    		Utils.platformLog("TABBAR", "create() result callback is NULL !");
    	}
    	NativeBar.create(NativeBar.TABBAR_TYPE, tabElements, tabBarProperties, result);
    }
    
    public void currentTabIndex(IMethodResult result) {
    	result.set(NativeBar.activeTab());
    }
    
    public void remove(IMethodResult result) {
    	NativeBar.remove();
    }
    
    public void setTabBadge(int tabIndex, String badge, IMethodResult result) {
    	// unsupported on Android
    }
    
    public void switchTab(int tabIndex, IMethodResult result) {
    	NativeBar.switchTab(tabIndex);
    }
    
    public void isCreated(IMethodResult result) {
    	result.set(NativeBar.isStarted());
    }
    
    public void removeTab(int tabIndex, IMethodResult result) {
    	//unsupported
    }

}

File: lib/commonAPI/coreapi/ext/platform/iphone/impl/NativeTabbarSingleton.m

#import "NativeTabbarSingleton.h"

#import "logging/RhoLog.h"
#undef DEFAULT_LOGCATEGORY
#define DEFAULT_LOGCATEGORY "NativeTabbar"


extern void rho_create_tabbar(NSArray* items, NSDictionary* properties, id<IMethodResult>callback);
extern BOOL nativetabbar_started();
extern void remove_native_tabbar();
extern void native_tabbar_set_tab_badge(int index,char *val);
extern int native_tabbar_get_current_tab();
extern void native_tabbar_switch_tab(int index);

@implementation NativeTabbarSingleton



-(void) create:(NSArray*)tabElements tabBarProperties:(NSDictionary*)tabBarProperties methodResult:(id<IMethodResult>)methodResult {
    if (![tabElements isKindOfClass:[NSArray class]]) {
        RAWLOG_ERROR("Tabbar.create ERROR - tabElements parameter should be Array !");
        return;
    }
    rho_create_tabbar(tabElements, tabBarProperties, methodResult);
}

-(void) currentTabIndex:(id<IMethodResult>)methodResult {
    [methodResult setResult:[NSNumber numberWithInt:native_tabbar_get_current_tab()]];
}

-(void) remove:(id<IMethodResult>)methodResult{
    remove_native_tabbar();
}

-(void) setTabBadge:(int)tabIndex badge:(NSString*)badge methodResult:(id<IMethodResult>)methodResult{
    native_tabbar_set_tab_badge(tabIndex, (char*)[badge UTF8String]);
}

-(void) switchTab:(int)tabIndex methodResult:(id<IMethodResult>)methodResult{
    native_tabbar_switch_tab(tabIndex);
}

-(void) isCreated:(id<IMethodResult>)methodResult {
    [methodResult setResult:[NSNumber numberWithBool:nativetabbar_started()]];
}

-(void) removeTab:(int)tabIndex methodResult:(id<IMethodResult>)methodResult{
    // unsupported
}



@end
